articles/architecture-scenarios/b2c/authentication.md

```diff
@@ -0,0 +1,120 @@
+
+::: note
+Checking for a login session by redirecting to Auth0 can be really helpful for your application, but if this will result in a lot of requests it is important to employ some sort of throttling mechanism to avoid latency and/or rate limiting.
+:::
+
+### Deep linking to protected endpoints
+
+There are a variety of reasons why someone might link directly to a particular page within your application that is only accessible by authenticated users. If this is possible for your application you should automatically redirect your user to Auth0 if they are not authenticated. Once they authenticate and the authorization server returns them to your application, you can [redirect them](/users/guides/redirect-users-after-login) to where they intended to go in the first place.
+
+::: note
+Most modern authentication frameworks support execution of middleware for redirecting to an authorization server such as Auth0. Ensure yours is configurable, can check expirations, and (for confidential clients) supports refresh tokens.
+:::
+
+### Authenticating the user
+
+Authentication is the process of verifying who a user is (that they are who they say they are). The result of authentication in this context is an ID Token. This token contains information about the user and should only be able to be obtained if the user authenticates using one or multiple factors as defined by the authorization server (the most common form being user id and password as discussed here). There are a few things you may also need to consider in addition to obtaining an ID Token:
+
+* Do we also need an Access Token in order to call a shared API?
+* Is our application a Native application (mobile or desktop) or do I need a Refresh Token?
+    * If Yes, for more information see Authorization Code Grant with PKCE
+    * If No, for more information see Implicit Grant
+
+::: warning
+Before you go live, you should ensure that only the grants that you are using for each application [are enabled in your configuration for the client](/applications/guides/update-grant-types-dashboard).
+:::
+
+### Implicit Grant
+
+If all your application needs is the ID Token and you are using a browser-based application, then you can always use the Implicit Grant to get your ID Token. This is a simple authentication flow and should be supported by your SDK (depending on the language you are developing in).
+
+If you need a refresh token so that you can re-check whether any of the information about the user has changed, then you must use the [authorization code grant](/api-auth/tutorials/authorization-code-grant).
```

Comment:
```suggestion
If you need a [Refresh Token](/tokens/refresh-token/current) so that you can obtain a new Access Token or ID Token without having to re-authenticate the user, then you must use the [authorization code grant](/api-auth/tutorials/authorization-code-grant).
```